sklearn/feature_selection/rfe.py

```diff
@@ -166,15 +178,19 @@ def _fit(self, X, y, step_score=None):
         if step_score:
             self.scores_ = []
 
+        step_one_threshold = (self.step_one_threshold if
+                              self.step_one_threshold is not None else 0)
```

Comment:
Should we be handling a float here too?

Reply:
Sure why not, we can also make it an int as well.  Meaning change the step rate to a different float or int at the threshold that corresponds to fewer features being removed per step.  This behavior wasn't mentioned in the Guyon SVM-RFE paper but it doesn't harm I guess (they only mentioned having the behavior where you remove more features at the beginning in a reducing step and at some number of remaining features based on your particular problem and dataset you start stepping by 1)

Reply:
@jnothman I updated code and doc based on suggestions above, except still don't understand the suggestion here for `try_all_below` or `step_if_at_least`?  I think maybe a slight misunderstanding.

Suppose you have a dataset with 3000 features.  When specifying step=100 (or a float) and step_one_threshold=250 then this is what will happen:

3000, 2900, 2800, ... , 400, 300, 250, 249, 248, 247, ... , n_features_to_select

The goal is to step by the specified (step=) larger number of features until you get to the specified threshold number of features where it will start stepping 1 at a time until it gets to n_features_to_select.  You can also see to that the way I wrote the code it won't ever step over the threshold no matter how larger the step size.

Reply:
If `step_one_threshold` is clearer to you, fine. What I mean by `try_all_below` is that from 250 down, all feature set sizes will be tried. Similarly, `step_if_at_least` indicates that if the feature set size is smaller than that value, `step` is disregarded.

Reply:
I’m not super happy with step_one_threshold either. But also feel the two suggestions could also be confusing to users.  try_all_below makes it seem like it’s going to try but could fail when in actually it will always do all below. step_if_at_least can be hard for people to understand because I ask myself step what if it at least?